From faed686a1a1aa8bec4fd0aa8832bf2b24af3a69d Mon Sep 17 00:00:00 2001
From: Haiying Wang <Haiying.Wang@freescale.com>
Date: Tue, 12 Nov 2013 12:03:33 -0500
Subject: [PATCH 341/429] fsl_qman: use physical cpu id to set/update stash
 setting

Signed-off-by: Haiying Wang <Haiying.Wang@freescale.com>
Change-Id: Ide019224408850e4d8e359445fc35eb7f4e77018
Reviewed-on: http://git.am.freescale.net:8181/6628
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Geoff Thorpe <Geoff.Thorpe@freescale.com>
Reviewed-by: Thomas Trefny <Tom.Trefny@freescale.com>
[origin patch is from QorIQ-SDK-V1.5-20131219-yocto]
Signed-off-by: Yang Wei <Wei.Yang@windriver.com>
---
 drivers/staging/fsl_qbman/qman_driver.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/qman_driver.c b/drivers/staging/fsl_qbman/qman_driver.c
index 2428dab..071805b 100644
--- a/drivers/staging/fsl_qbman/qman_driver.c
+++ b/drivers/staging/fsl_qbman/qman_driver.c
@@ -594,7 +594,7 @@ static struct qman_portal *init_pcfg(struct qm_portal_config *pcfg)
 	struct qman_portal *p;
 
 	pcfg->iommu_domain = NULL;
-	portal_set_cpu(pcfg, pcfg->public_cfg.cpu);
+	portal_set_cpu(pcfg, get_hard_smp_processor_id(pcfg->public_cfg.cpu));
 	p = qman_create_affine_portal(pcfg, NULL);
 	if (p) {
 		u32 irq_sources = 0;
@@ -688,7 +688,8 @@ static void qman_online_cpu(unsigned int cpu)
 		pcfg = qman_get_qm_portal_config(p);
 		if (pcfg) {
 			irq_set_affinity(pcfg->public_cfg.irq, cpumask_of(cpu));
-			qman_portal_update_sdest(pcfg, cpu);
+			qman_portal_update_sdest(pcfg,
+					get_hard_smp_processor_id(cpu));
 		}
 	}
 }
-- 
1.7.5.4

